### PR TITLE
Ignore czech special characters during codeword comparison

### LINKED ISSUE #290: Upravit kontrolu zadaneho codeword
Stavajici porovnavani:
- Codeword je prijaty bez ohledu na velka a mala pismena
Pozadovane porovnavani
- Codeword je prijaty bez ohledu na velka a mala pismena a bezohledu na ceskou diakritiku

Priklady:
Pro codeword "Šmoulové" budou toto jedny ze spravych zadani:
- Šmoulove
- smoulove
- sMoulové
- atd.

### PR DESCRIPTION
**Pull request by mjanousek**
fixes #290

### CODE REVIEW

File: src/main/java/dk/cngroup/lentils/service/CypherService.java [hunk 1]
```diff
@@ -25,37 +23,35 @@ public void setStatusService(final StatusService statusService) {
...
         this.cypherRepository = cypherRepository;
-        this.teamService = teamService;
-        this.hintTakenService = hintTakenService;
     }
 
     public Cypher save(final Cypher cypher) {
         return cypherRepository.save(cypher);
     }
 
-    public List<Hint> getHintsForStage(final Integer stage) {
```
Comment (daweedek2) [inline src/main/java/dk/cngroup/lentils/service/CypherService.java]: co se stalo s touto metodou?

File: src/main/java/dk/cngroup/lentils/service/CypherService.java [hunk 2]
```diff
@@ -74,18 +70,6 @@ public void saveAll(final List<Cypher> cyphers) {
         return cypherRepository.findAllByOrderByStageAsc();
     }
 
-    public int getScore(final Cypher cypher, final Team team) {
```
Comment (daweedek2) [inline src/main/java/dk/cngroup/lentils/service/CypherService.java]: co se stalo s touto metodou?

File: src/main/java/dk/cngroup/lentils/service/CypherService.java [hunk 3]
```diff
@@ -74,18 +70,6 @@ public void saveAll(final List<Cypher> cyphers) {
...
-        int statusScore = statusService.getStatusScore(team, cypher);
-        int hintScore = hintTakenService.getHintScore(team, cypher);
-        return statusScore - hintScore;
-    }
-
-    public int getScore(final Long cypherId, final Long teamId) {
```
Comment (daweedek2) [inline src/main/java/dk/cngroup/lentils/service/CypherService.java]: co se stalo s touto metodou?

File: src/test/java/dk/cngroup/lentils/repository/UserRepositoryTest.java [hunk 4]
```diff
@@ -81,6 +83,16 @@ public void addTeamGeneratesProperUsername() {
         Assert.assertTrue(isUserInDb(GENERATED_USERNAME));
     }
 
+    @Test
+    public void t1() {
```
[the remark this answers is not in the document]
Comment (mjanousek) [reply src/test/java/dk/cngroup/lentils/repository/UserRepositoryTest.java]: Sakra to tam vubec nema byt. :D diky ;)

File: src/main/java/dk/cngroup/lentils/service/CypherService.java (hunk 3 above)
Comment (mjanousek) [reply src/main/java/dk/cngroup/lentils/service/CypherService.java]: Byla nepouzita.

File: src/main/java/dk/cngroup/lentils/service/CypherService.java (hunk 2 above)
Comment (mjanousek) [reply src/main/java/dk/cngroup/lentils/service/CypherService.java]: Byla nepouzita

File: src/main/java/dk/cngroup/lentils/service/CypherService.java (hunk 1 above)
Comment (mjanousek) [reply src/main/java/dk/cngroup/lentils/service/CypherService.java]: Byla nepouzita

File: src/test/java/dk/cngroup/lentils/service/CypherServiceTest.java [hunk 5]
```diff
@@ -29,26 +31,49 @@
...
+
+    @Test
+    public void checkCodewordWithSpecialCzechCharactersTest() {
+        Cypher cypher = getCypherForStage(CODEWORD_WITHOUT_CZECH_SPECIAL_CHARACTERS);
+
+        assertTrue(service.checkCodeword(cypher, CODEWORD_WITH_CZECH_SPECIAL_CHARACTERS));
+    }
+
+    private Cypher getCypherForStage(Integer stage, String codeword) {
+        Cypher cypher = new Cypher("blabla", stage, new Point(0,0), codeword, TEST_MAP_ADDRESS);
```
Comment (daweedek2) [inline src/test/java/dk/cngroup/lentils/service/CypherServiceTest.java]: nazev sifry "blabla" by mohl byt definovany v promenne

File: src/test/java/dk/cngroup/lentils/repository/UserRepositoryTest.java [hunk 6]
```diff
@@ -4,6 +4,7 @@
 import dk.cngroup.lentils.entity.Team;
 import dk.cngroup.lentils.entity.User;
 import dk.cngroup.lentils.service.TeamService;
+import dk.cngroup.lentils.util.UsernameUtils;
```
Comment (crlikcngroup) [inline src/test/java/dk/cngroup/lentils/repository/UserRepositoryTest.java]: Unused import.

File: src/test/java/dk/cngroup/lentils/repository/UserRepositoryTest.java [hunk 7]
```diff
@@ -12,6 +13,7 @@
 import org.springframework.test.context.junit4.SpringRunner;
 import org.springframework.transaction.annotation.Transactional;
 
+import java.text.Normalizer;
```
Comment (crlikcngroup) [inline src/test/java/dk/cngroup/lentils/repository/UserRepositoryTest.java]: Unused import.

**Pull request merged**